Stepic/PersonalDeadlineLocalNotificationContentProvider.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  PersonalDeadlineLocalNotificationContentProvider.swift
+//  Stepic
+//
+//  Created by Ivan Magda on 15/10/2018.
+//  Copyright © 2018 Alex Karpov. All rights reserved.
+//
+
+import Foundation
+import UserNotifications
+
+final class PersonalDeadlineLocalNotificationContentProvider: LocalNotificationContentProvider {
+    var title: String {
+        return "\(course.title)"
+    }
+
+    var body: String {
+        if #available(iOS 10.0, *) {
+            return NSString.localizedUserNotificationString(
```

Comment:
Зачем, если можно сократить код, используя только NSLocalizedString?

Reply:
Для чего тогда вообще сделана эта localizedUserNotificationString? Не может же быть, что просто так продублировали NSLocalizedString?

Reply:
Я где-то читал, что когда нотификация собирается отобразится, то UNNotificationContent ждёт пока строка не загрузится и только после этого отображается, если использовать localizedUserNotificationString. Кажется, что просто оптимизация, ничего страшного не будет если использовать NSLocalizedString.

В документации про это не написано, в каком-то бложеке читал.

Reply:
Ответ документации на то, почему стоит использовать `NSString.localizedUserNotificationString` при создании `UNMutableNotificationContent`

https://developer.apple.com/documentation/usernotifications/unmutablenotificationcontent

The `localizedUserNotificationString(forKey:arguments:)` method delays the loading of the localized string until the notification is delivered. Thus, if the user changes language settings before a notification is delivered, the alert text is updated to the user’s current language instead of the language that was set when the notification was scheduled.

Для нас это оказалось адекватным аргументом.